src/WebJobs.Script.WebHost/Filters/ArmExtensionResourceFilter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Azure.WebJobs.Script.WebHost.Properties;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Filters
+{
+    /// <summary>
+    /// Resource filter used to ensure secrets aren't returned for GET requests made via the Functions ARM extension
+    /// API (hostruntime), unless properly authorized.
+    /// </summary>
+    public sealed class ArmExtensionResourceFilter : IAsyncResourceFilter
+    {
+        public async Task OnResourceExecutionAsync(ResourceExecutingContext context, ResourceExecutionDelegate next)
+        {
+            bool isArmExtensionRequest = !string.IsNullOrEmpty(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresARMExtensionsRouteHeader));
+            bool isOwnerOrCoAdmin = string.Equals(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresClientAuthorizationSourceHeader), "legacy", StringComparison.OrdinalIgnoreCase);
+            bool isRoleBasedContributor = string.Equals(context.HttpContext.Request.GetHeaderValueOrDefault(ScriptConstants.AntaresRoleBasedContributorHeader), "1");
```

Comment:
@suwatch this header will only be set to 1 if the request has been properly RBAC authorized? When testing via ArmClient I don't see this header set.